File: weather.rb
require 'net/http'

def getWeather
  url = "http://mat1.qq.com/weather/inc/minisite2_252.js";
  ret = Net::HTTP.get(URI(url));
  return ret;
end

def matchSTR(data)
  ec = Encoding::Converter.new("gbk", "UTF-8")
  str = ec.convert(data)
  str = /([^a-zA-Z0-9=_"\s]).*(?=")/.match(str)
  return str[0];
end

def sendFetion(str)
  url = "http://quanapi.sinaapp.com/fetion.php?u=15901965812&p=hb25524289&to=15901965812&m="+URI.escape(str);
  ret = Net::HTTP::get(URI(url))
end

data  = getWeather
str = matchSTR(data);
sendFetion(str);

File: weather.php
<?php
require 'feixin.php';

$data  = getWeather();
$str = matchSTR($data);
sendFetion($str);

 function matchSTR($data){
     preg_match('/".*"/', $data, $matches);
     $str = trim( iconv('gbk','UTF-8',$matches[0]), '"' );
     return $str;
 }

function getWeather(){
  $ch = curl_init();
  $url = "http://mat1.qq.com/weather/inc/minisite2_252.js";
  curl_setopt($ch, CURLOPT_URL, $url);
  curl_setopt($ch, CURLOPT_HEADER, 0);
  curl_setopt($ch,CURLOPT_RETURNTRANSFER,true);
  $ret = curl_exec($ch);
  curl_close($ch);
  return $ret;
}
